### PR TITLE
Collection of fixes from last round if reviews in the feature metrics PR

### PR DESCRIPTION
**Pull request by purbon**
Hi,
  this actually makes sure to fix a few of the concerns coming out of the last round of reviews in the metrics api, in detail:

* make sure puma version >= 2.16 (same as in the http input)
* add http-host option to the web server.
* fix typos and wrongly removed lines.

@ph what do you think?

/cc @jsvd

### CODE REVIEW

File: logstash-core/lib/logstash/runner.rb [hunk 1]
```diff
@@ -82,7 +82,11 @@ class LogStash::Runner < Clamp::Command
     I18n.t("logstash.runner.flag.auto_reload"),
     :attribute_name => :auto_reload, :default => false
 
-  option ["-ap", "--http-port"], "WEB_API_HTTP_PORT",
+  option ["--http-host"], "WEB_API_HTTP_HOST",
+    I18n.t("logstash.web_api.flag.http_host"),
+    :attribute_name => :web_api_http_host, :default => "0.0.0.0"
```
Comment (suyograo) [inline logstash-core/lib/logstash/runner.rb]: Can we use `127.0.0.1` as default instead of `0.0.0.0`

File: Gemfile [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 source "https://rubygems.org"
 gem "logstash-core", "3.0.0.dev", :path => "./logstash-core"
+# gem "logstash-core-event", "3.0.0.dev", :path => "./logstash-core-event"
```
Comment (ph) [inline Gemfile]: I think this is not the last time we are bitten by this line.

File: logstash-core/lib/logstash/runner.rb (hunk 1 above)
Comment (ph) [inline logstash-core/lib/logstash/runner.rb]: +1

**Comment by ph** [comment]
Only the minor comments concerning 127.0.0.1 and it lgtm.
We should bind to localhost only by default and not all the interface.

File: logstash-core/lib/logstash/runner.rb (hunk 1 above)
Comment (purbon) [inline logstash-core/lib/logstash/runner.rb]: sure

File: Gemfile (hunk 2 above)
Comment (purbon) [inline Gemfile]: yeah, I do really think we should remove it ....

**Pull request closed without merging**